src/main/java/com/ss/training/lms/controller/AdminBookController.java

```diff
@@ -0,0 +1,186 @@
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+/**
+ * @author Trevor Huis in 't Veld
+ */
+@RestController
+@RequestMapping(path="lms/admin/book")
+public class AdminBookController {
+
+    @Autowired
+    AdminBookService bookService;
+
+    @Autowired
+    AdminGenreService genreService;
+
+    @Autowired
+    AdminAuthorService authorService;
+
+    /**
+	 * @param book
+	 * @return
+	 */
+	@PostMapping
+	public ResponseEntity<Book> createBook(@RequestBody Book book) {
+        HttpStatus status = HttpStatus.BAD_REQUEST;
+        if (book == null || book.getTitle() == null || book.getTitle().length() > 45 || book.getPublisherId() == null)
+			return new ResponseEntity<Book>(book, status);
+		try {
+			bookService.createBook(book);
+			status = HttpStatus.CREATED;
+		} catch (ClassNotFoundException | SQLException e) {
+			status = HttpStatus.INTERNAL_SERVER_ERROR;
+		}
+		return new ResponseEntity<Book>(book, status);
+    }
+
+    /**
+     * 
+     * @param bookId
+     * @return
+     */
+	@RequestMapping(path="/{id}")
+	public ResponseEntity<Book> readBook(@PathVariable int bookId) {
+		Book book = null;
+		HttpStatus status = HttpStatus.OK;
+		try {
+			book = bookService.readBook(bookId);
+			if (book == null) 
+				status = HttpStatus.NOT_FOUND;
+		} catch ( SQLException e) {
+			status = HttpStatus.INTERNAL_SERVER_ERROR;
+		}
+		return new ResponseEntity<Book>(book, status);
+	}
+
+    /**
+     * 
+     * @return
+     */
+	@RequestMapping
+	public ResponseEntity<List<Book>> readBooks() {
+		List<Book> books = null;
+		HttpStatus status = HttpStatus.OK;
+		try {
+			books = bookService.readAllBooks();
+			if (books == null) // no Books exist in the database
+				status = HttpStatus.NO_CONTENT;
+		} catch (ClassNotFoundException | SQLException e) {
+			status = HttpStatus.INTERNAL_SERVER_ERROR;
+		}
+		return new ResponseEntity<List<Book>>(books, status);
+	}
+
+    /**
+     * 
+     * @param bookId
+     * @return
+     */
+	@DeleteMapping(path="/{id}")
+	public ResponseEntity<Book> deleteBook(@PathVariable int bookId) {
+		Book book = null;
+		HttpStatus status = HttpStatus.OK;
+		try {
+			book = bookService.readBook(bookId);
+			if (book == null) 
+				status = HttpStatus.NOT_FOUND;
+			else
+				bookService.deleteBook(book);
+		} catch (ClassNotFoundException | SQLException e) {
+			status = HttpStatus.INTERNAL_SERVER_ERROR;
+		}
+		return new ResponseEntity<Book>(book, status);
+	}
+
+    /**
+     * 
+     * @param book
+     * @return
+     */
+	@PutMapping
+	public ResponseEntity<Book> updateBook(@RequestBody Book book) {
+		HttpStatus status = HttpStatus.BAD_REQUEST;
+		if (book == null || book.getTitle().length() > 45)
```

Comment:
We should also make sure the title is non-null here, since the schema doesn't allow books to have null titles.